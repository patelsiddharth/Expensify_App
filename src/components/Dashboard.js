import React from 'react';
import ExpenseList from './ExpenseList';
import ExpenseListFilters from './ExpenseListFilters';

const ExpenseDashboard = ()=> (
    <div>
        This is Expensify App DashBoard Page
        <ExpenseListFilters />
        <ExpenseList />
    </div>
)

export default ExpenseDashboard;