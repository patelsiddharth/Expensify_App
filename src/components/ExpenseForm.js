import React from 'react';
import moment from 'moment';
import { SingleDatePicker } from 'react-dates';

export default class ExpenseForm extends React.Component
{
    constructor(props)
    {
        super(props);

        this.state = {
            description : props.expense ? props.expense.description : '',
            note : props.expense ? props.expense.note :'',
            amount : props.expense ? props.expense.amount.toString() : '',
            createdAt : props.expense ? moment(props.expense.createdAt) : moment(),
            calendarFocused : false,
            error : false
        }
    }

    onDescriptionChange = (e) => {
        const description = e.target.value;
        this.setState(() => ({ description }))
    };

    onNoteChange = (e) => {
        const note = e.target.value;
        this.setState( () => ({ note }));
    };

    onAmountChange = (e) => {
        const amount = e.target.value;
        if(amount.match(/^\d*(\.\d{0,2})?$/))
        {
            this.setState( () => ({amount}))
        }
    };

    onDateChange = (createdAt) => {
        if(createdAt)
        {
            this.setState( () => ({createdAt}))
        }
    }

    onFocusChange = ({ focused }) => {
        this.setState( () => ({ calendarFocused : focused }))
    }
    onFormSubmit = (e) => {
        e.preventDefault();
        if(!this.state.description || !this.state.amount)
        {
            this.setState( () => ({error : true}))
        }
        else
        {
            this.setState( () => ({error : false}))
            this.props.onSubmit({
                description : this.state.description,
                amount : parseFloat(this.state.amount),
                createdAt : this.state.createdAt.valueOf(),
                note : this.state.note
            })
        }
    }

    render()
    {
        return (
            <form className="form" onSubmit={this.onFormSubmit}>
                {this.state.error && <p className="form__error">Please provide description and amount</p>}
                <input className="text-input"
                        type='text' placeholder='Description' autoFocus 
                        value={this.state.description}
                        onChange={this.onDescriptionChange}
                />
                <input className="text-input" 
                        type='number' placeholder='Amount'
                        value={this.state.amount}
                        onChange={this.onAmountChange}
                />
                <SingleDatePicker 
                    className="date-time"
                    date={this.state.createdAt}
                    onDateChange = {this.onDateChange}
                    focused = {this.state.calendarFocused}
                    onFocusChange = {this.onFocusChange}
                    numberOfMonths={1}
                    isOutsideRange={() => false}
                />
                <textarea className="textarea"
                            placeholder='Add a note for your expense'
                            value={this.state.note}
                            onChange={this.onNoteChange}>
                </textarea>
                <button className="button">Add Expense</button>
            </form>
        )
    }
}