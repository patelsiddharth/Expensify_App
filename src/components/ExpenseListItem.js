import React from 'react';
import { Link } from 'react-router-dom';
import moment from 'moment';
import numeral from 'numeral';

numeral.register('locale', 'in', {
    delimiters: {
        thousands: ' ',
        decimal: '.'
    },
    currency: {
        symbol: 'â‚¹'
    }
});

numeral.locale('in');

const ExpenseListItem = ({id, description, amount, createdAt}) => (
    <div>
        <Link to={`/edit/${id}`}>
            <h3>{description}</h3>
        </Link>
        <p>
            {numeral(amount).format('$0,0.00')} 
            - 
            {moment(createdAt).format('MMMM Do, YYYY')}
        </p>
    </div>
)

export default ExpenseListItem;