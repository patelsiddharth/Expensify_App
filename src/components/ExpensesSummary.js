import React from 'react';
import {connect} from 'react-redux';
import {Link} from 'react-router-dom';
import numeral from 'numeral';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';

export const ExpensesSummary = ({expenseCount, expenseTotal}) => {
    const isExpenseSingular = expenseCount === 1 ? 'expense' : 'expenses';
    const formattedTotal = numeral(expenseTotal).format('$0,0.00');
    return (
        <div className="page-header">
            <div className="content-container">
                <h1 className="page-header__title">
                    Viewing <b>{expenseCount}</b> {isExpenseSingular} totalling <b>{formattedTotal}</b>
                </h1>
                <div className="page-header__actions">
                    <Link className="button" to='/create'>Add Expanse</Link>
                </div>
            </div>
        </div>
    )
}

const mapStateToProps = (state) => {
    const visibleExpenses = selectExpenses(state.expenses, state.filters);
    return {
        expenseCount : visibleExpenses.length,
        expenseTotal : selectExpensesTotal(visibleExpenses)
    }
}

export default connect(mapStateToProps)(ExpensesSummary);