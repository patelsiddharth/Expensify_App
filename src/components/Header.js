import React from 'react';
import {NavLink} from 'react-router-dom';

const Header = () => (
    <header>
        <h1>Expensify App</h1>
        <NavLink to='/' activeClassName='is-active' exact>DashBoard</NavLink>
        <NavLink to='/create' activeClassName='is-active'>Add Expanse</NavLink>
        <NavLink to='/edit' activeClassName='is-active'>Edit Expanse</NavLink>
        <NavLink to='/help' activeClassName='is-active'>Help Page</NavLink>
    </header>
)

export default Header;