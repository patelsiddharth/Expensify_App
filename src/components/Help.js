import React from 'react';

const Help = ()=> (
    <div>
        This is Expensify App Help Page
    </div>
)

export default Help;