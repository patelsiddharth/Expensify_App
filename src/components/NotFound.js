import React from 'react';
import {Link} from 'react-router-dom';

const NotFound = ()=> (
    <div>
        <p>404 !!! Page Not Found.</p>
        <Link to='/'>Go to Home Page</Link>
    </div>
)

export default NotFound;