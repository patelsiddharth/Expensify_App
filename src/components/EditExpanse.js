import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { startEditExpense, startRemoveExpense } from '../actions/expenses';
import { Link } from 'react-router-dom';

export class EditExpanse extends React.Component
{
    onSubmit = (expense) => {
        this.props.startEditExpense(this.props.expense.id, expense);
        this.props.history.push('/dashboard');
    }

    onRemove = (e) => {
        this.props.startRemoveExpense({id : this.props.expense.id});
        this.props.history.push('/dashboard');
    }

    render()
    {
        return (
            <div>
                <Link to='/dashboard'>Back to Dashboard</Link>
                <ExpenseForm 
                    expense={this.props.expense}
                    onSubmit = {this.onSubmit}
                />
                <button onClick={this.onRemove}>Remove Item</button>
            </div>
        );
    }
}

const mapStateToProps = (state, props) => ({
    expense : state.expenses.find(expense => expense.id === props.match.params.id)
})

const mapDispatchToProps = (dispatch, props) => ({
    startEditExpense : (id, expense) => dispatch(startEditExpense(id, expense)),
    startRemoveExpense : (data) => dispatch(startRemoveExpense(data))
})

export default connect(mapStateToProps, mapDispatchToProps)(EditExpanse);

// const EditExpanse = (props) => {
//     return (
//         <div>
//             <Link to='/'>Back to Dashboard</Link>
//             <ExpenseForm 
//                 expense={props.expense}
//                 onSubmit = { expense => {
//                     props.dispatch(editExpense(props.expense.id, expense));
//                     props.history.push('/');
//                 }}
//             />
//             <button onClick={ e => {
//                 props.dispatch(removeExpense({id : props.expense.id}))
//                 props.history.push('/');
//             }}>Remove Item</button>
//         </div>
//     );
// };