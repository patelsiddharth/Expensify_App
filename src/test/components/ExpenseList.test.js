import React from "react";
import {shallow} from 'enzyme';
import { ExpenseList } from '../../components/ExpenseList';
import expense from '../fixtures/expenses';

test('Should render ExpenseList with expenses', () => {
    const wrapper = shallow(<ExpenseList expenses={expense}/>);
    expect(wrapper).toMatchSnapshot();
})

test('Should render No Expense found meesage when expenses is empty', () => {
    const wrapper = shallow(<ExpenseList expenses={[]}/>);
    expect(wrapper).toMatchSnapshot();
})