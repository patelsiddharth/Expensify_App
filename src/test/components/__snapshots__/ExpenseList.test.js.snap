// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render ExpenseList with expenses 1`] = `
<div
  className="content-container"
>
  <div
    className="list-header"
  >
    <div
      className="show-for-mobile"
    >
      Expenses
    </div>
    <div
      className="show-for-desktop"
    >
      Expense
    </div>
    <div
      className="show-for-desktop"
    >
      Amount
    </div>
  </div>
  <div
    className="list-body"
  >
    <ExpenseListItem
      amount={45}
      createdAt={0}
      description="Gum"
      id="1"
      key="1"
      note=""
    />
    <ExpenseListItem
      amount={123}
      createdAt={-345600000}
      description="Coffee"
      id="2"
      key="2"
      note=""
    />
    <ExpenseListItem
      amount={3250}
      createdAt={345600000}
      description="Rent"
      id="3"
      key="3"
      note=""
    />
  </div>
</div>
`;

exports[`Should render No Expense found meesage when expenses is empty 1`] = `
<div
  className="content-container"
>
  <div
    className="list-header"
  >
    <div
      className="show-for-mobile"
    >
      Expenses
    </div>
    <div
      className="show-for-desktop"
    >
      Expense
    </div>
    <div
      className="show-for-desktop"
    >
      Amount
    </div>
  </div>
  <div
    className="list-body"
  >
    <div
      className="list-item list-item__message"
    >
      <span>
        No Expenses
      </span>
    </div>
  </div>
</div>
`;
